### PR TITLE
Fixup AzDO client construction in DI

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Resolves https://github.com/dotnet/arcade-services/issues/3718

Found by https://dev.azure.com/dnceng/internal/_build/results?buildId=2492728&view=results

DI is complaining about:
```
System.InvalidOperationException: Unable to activate type 'Microsoft.DotNet.DarcLib.AzureDevOpsClient'. The following constructors are ambiguous:
Void .ctor(Maestro.Common.AzureDevOpsTokens.IAzureDevOpsTokenProvider, Microsoft.DotNet.DarcLib.Helpers.IProcessManager, Microsoft.Extensions.Logging.ILogger`1[Microsoft.DotNet.DarcLib.AzureDevOpsClient])
Void .ctor(Maestro.Common.AzureDevOpsTokens.IAzureDevOpsTokenProvider, Microsoft.DotNet.DarcLib.Helpers.IProcessManager, Microsoft.Extensions.Logging.ILogger)
```

I couldn't find a way to remove one of the constructors and not break one of the services or Darc, so instead opted for manually constructing the AzDO client instance in DI.

Manually tested the command locally. Test build: https://dev.azure.com/dnceng/internal/_build?definitionId=252&_a=summary

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/Darc/Operations/Operation.cs [hunk 1]
```diff
@@ -66,9 +66,19 @@ protected Operation(ICommandLineOptions options, IServiceCollection? services =
...
-        services.TryAddSingleton<IAzureDevOpsClient, AzureDevOpsClient>();
         services.TryAddSingleton<IAzureDevOpsTokenProvider, AzureDevOpsTokenProvider>();
+        services.TryAddSingleton(s =>
+            new AzureDevOpsClient(
+                s.GetRequiredService<IAzureDevOpsTokenProvider>(),
+                s.GetRequiredService<IProcessManager>(),
+                s.GetRequiredService<ILogger>())
+        );
+        services.TryAddSingleton<IAzureDevOpsClient>(s =>
+            new AzureDevOpsClient(
```
[the remark this answers is not in the document]
Comment (oleksandr-didyk) [reply src/Microsoft.DotNet.Darc/Darc/Operations/Operation.cs:77]: Oh, true, thanks for the tip

**Review by dkurepa — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@oleksandr-didyk @dkurepa I'm not sure if this fixed the issue. https://dev.azure.com/dnceng/internal/_build/results?buildId=2493151&view=logs&j=29d4eac0-e772-56a4-ad0d-2d727d908ceb&t=25f6d0f0-c688-5dd7-fabf-104a8f0cb3a8 is still failing which already got the arcade-services dependency update.

**Comment by ViktorHofer** [comment]
OK looks like arcade-services's "production" build is failing: https://dev.azure.com/dnceng/internal/_build?definitionId=252

**Comment by oleksandr-didyk** [comment]
The fix will not take effect until its in production; Matt committed the change to prod yesterday, but its build is getting stuck. Once we figure it out and the Darc version Maestro is referencing is updated, the promotion should get unblocked

**Pull request merged**